pkg/infra/usagestats/service/usage_stats.go

```diff
@@ -118,6 +119,12 @@ func (uss *UsageStats) GetUsageReport(ctx context.Context) (usagestats.Report, e
 		return report, err
 	}
 
+	// send usage stats exposed by plugins in CollectMetrics
+	customPluginStats := uss.customPluginStats(ctx)
+	for k, v := range customPluginStats {
+		metrics[k] = v
+	}
+
```

Comment:
Rather than adding special-purpose code here could we use the RegisterMetricsFunc approach like we do in live stats?

https://github.com/grafana/grafana/blob/main/pkg/services/live/live.go#L1384-L1412

Reply:
@DanCech i attempted that, but wire complained about a circular dependency because usage stats includes the plugin store

Reply:
that is annoying, the dependency is there for 2 purposes, to get some counts that could be reported by the plugin store directly and also to filter the list of datasources based on the plugin status which is trickier.  One option might be to factor the external metrics registration code out into a separate service that plugins could depend on to register its metric function(s) and usagestats could depend on to get the list of registered metrics functions